core/src/main/java/org/elasticsearch/index/query/PrefixQueryParser.java

```diff
@@ -73,7 +73,7 @@ public Query parse(QueryParseContext parseContext) throws IOException, QueryPars
                         if ("_name".equals(currentFieldName)) {
                             queryName = parser.text();
                         } else if ("value".equals(currentFieldName) || "prefix".equals(currentFieldName)) {
-                            value = parser.objectBytes();
+                            value = parser.textOrNull();
```

Comment:
same as in the regex PR, I wonder if we should have parser.text() or parser.textOrNull()